pyprep/prep_pipeline.py

```diff
@@ -56,14 +56,32 @@ def __init__(self, raw, prep_params, montage, ransac=True, random_state=None):
         self.raw = raw.copy()
         self.ch_names = self.raw.ch_names
         self.raw.set_montage(montage)
-        self.raw.pick_types(eeg=True, eog=False, meg=False)
+        self.raw_non_eeg = self.raw.copy()
+
+        ch_types = ["eeg", "meg", "stim", "eog", "ecg", "emg", "ref_meg",
+                    "misc", "resp", "syst", "seeg", "dipole", "gof",
+                    "bio", "ecog", "fnirs", "csd"]
```

Comment:
given that we work reasonably close with MNE, I wonder if it'd be good or bad to use private variable from MNE such as this one:

https://github.com/mne-tools/mne-python/blob/76fc838e954e70697f33cab0edbca61872ca1c23/mne/io/pick.py#L70

@larsoner do you perhaps have advice on this? We want a list of all channel types available in MNE.

Reply:
Perhaps the keys of the dict returned by `get_channel_type_constants()`

https://github.com/mne-tools/mne-python/blob/master/mne/io/pick.py#L18

Reply:
Problem would be that there are little discrepancies between the two:

the dict keys would be:
```
['bio', 'chpi', 'csd', 'dipole', 'ecg', 'ecog', 'eeg', 'emg', 'eog', 'exci', 'fnirs_od', 'fnirs_raw', 'gof', 'grad', 'hbo', 'hbr', 'ias', 'mag', 'misc', 'ref_meg', 'resp', 'seeg', 'stim', 'syst']
```
whereas the pick_types channel arguments are:
```
['bio', 'chpi', 'csd', 'dipole', 'ecg', 'ecog', 'eeg', 'emg', 'eog', 'exci', 'fnirs', 'gof', 'ias', 'meg', 'misc', 'ref_meg', 'resp', 'seeg', 'stim', 'syst']
```

So we would need little non-homogeneous corrections like:
- fnirs_od fnirs_raw to single fnirs
- delete grad hbo hbr
- replace mag with meg

Im not sure if this simplifies or complicates the stuff.

Another way is using inspect module:
```
ch_types = inspect.signature(mne.io.pick.pick_types)
ch_types = ch_types.parameters.items()
ch_types = list(ch_types)
ch_types = [x[0] for x in ch_types]
ch_types.pop(0) #remove info
ch_types = ch_types[:-3] #remove non channel parameters
```
Which Im also not sure if it is better than making the list like the initial approach

Reply:
Regarding the _first_rule private variable I dont know how could we extract it but if we manage to do it then the only change would be to add 'csd' to it.

Reply:
> Which Im also not sure if it is better than making the list like the initial approach

we can also stay with a list of strings instead of importing that list directly from MNE-Python ... but I wanted to check whether there is an easy way to directly import this --> it would decrease maintenance burden. 

IF we hardcode the list here, we'll have to update it every time something about these channel types changes in MNE

Reply:
how do you feel about the inspect module way? Although it is a bit ugly I find the following positive points:

- If mne changes then the pick_type method would have changed accordingly.
- I think that params positions will stay the same , that is , info im almost sure it will stay in front and the other non chan_type params will be the last three --> include, exclude , selection.

We could also extract them from raw.pick_types rather than io.pick , then we would get rid of the last 4.

Cons:
- ugly
- maybe unpythonic
- confusing for someone just reading the code ?
- breaks if non chan_type params are added or change position

Reply:
Maybe another way would be using the include exclude parameters of pick, assuming 'eeg' types are assigned correctly which the other method also needs

- obtain list of eeg chans --> non eeg channels = all - eeg chans
- include only non eeg_channels for copy/extraction

I think this may be the one. Currently coding the implementation.